hack/verify-generated-swagger-docs.sh

```diff
@@ -38,29 +38,57 @@ if [[ ! -x "$genswaggertypedocs" ]]; then
   exit 1
 fi
 
-DIFFROOT="${KUBE_ROOT}/pkg"
-TMP_DIFFROOT="${KUBE_ROOT}/_tmp/pkg"
-_tmp="${KUBE_ROOT}/_tmp"
-
-cleanup() {
-  rm -rf "${_tmp}"
+_tmpdir="$(kube::realpath "$(mktemp -d -t swagger-docs.XXXXXX)")"
+function swagger_cleanup {
+  rm -rf "${_tmpdir}"
 }
-trap "cleanup" EXIT SIGINT
+kube::util::trap_add swagger_cleanup EXIT
+
+# Copy the contents of the kube directory into the nice clean place
+_kubetmp="${_tmpdir}/src/k8s.io"
+mkdir -p "${_kubetmp}"
+# should create ${_kubetmp}/kubernetes
+git archive --format=tar --prefix=kubernetes/ "$(git write-tree)" | (cd "${_kubetmp}" && tar xf -)
```

Comment:
is this as expensive as it looks?

Reply:
Not really.. we use it in other places too where we want to copy and write the whole git tree for use in tests.

Reply:
ok. I'll defer to @BenTheElder for lgtm

Reply:
tar | untar is kinda expensive but one of the more reasonable options for copying to an isolated tempdir.

Reply:
(we also do this already as Christoph mentioned, it might be worth revisiting but I'd rather unblock for now)